### PR TITLE
Adding ability to edit, delete and add new relief techniques

### LINKED ISSUE #125: [Relief Techniques] Allow users to add new relief techniques

### LINKED ISSUE #146: Fix bug where data is not cleared from "add relief technique" textfields after relief technique is added

### PR DESCRIPTION
**Pull request by trunci**
## About the PR

Adding the ability to edit, delete and add new relief techniques. users are free to add anything they want or delete old ones.

## Testing
[screen-capture (1).webm](https://user-images.githubusercontent.com/26496730/201965443-beb52781-1b9b-4b31-9ba5-308f9ffe9af8.webm)

## Introduced Bugs

* Changing the name of an activity will not delete the old one
* There isn't a way to reset to the normal activities

### CODE REVIEW

File: lib/components/ReliefActivitiesMenu/ActivityButton.dart [hunk 1]
```diff
@@ -54,14 +68,207 @@ class _ActivityButtonState extends State<ActivityButton> {
...
+                                                .toList(),
+                                            style: const TextStyle(
+                                              color: AppColors.font,
+                                              fontFamily: 'MainFont',
+                                              fontWeight: FontWeight.w600,
+                                              fontSize: 18,
+                                            ),
+                                            decoration: const InputDecoration(
+                                                hintText:
+                                                    'Enter the mood for this activity',
```
Comment (AlexShor0103) [inline lib/components/ReliefActivitiesMenu/ActivityButton.dart:148]: Very Minor, but I think it'd be better to have this text be "Enter what mood this activity would help"

File: lib/components/ReliefActivitiesMenu/ActivityButton.dart [hunk 2]
```diff
@@ -54,14 +68,207 @@ class _ActivityButtonState extends State<ActivityButton> {
...
+                                                    'Enter the URL of the video of the Activity',
+                                                labelText: 'URL',
+                                                labelStyle: TextStyle(
+                                                  color: AppColors.font,
+                                                  fontFamily: 'MainFont',
+                                                  fontWeight: FontWeight.w600,
+                                                  fontSize: 17,
+                                                )),
+                                            onSaved: (String? value) {
+                                              URL = value ?? 'Activity\'s URL';
```
Comment (AlexShor0103) [inline lib/components/ReliefActivitiesMenu/ActivityButton.dart:203]: Would the URL link to the browser? And if so how would the user rate/ verify completion of the activity?

**Review by AlexShor0103 — APPROVED** [review]
LGTM! Just wondering how the URL would work on the relief techniques but other than that it's a solid addition

File: lib/components/ReliefActivitiesMenu/ActivityButton.dart (hunk 1 above)
Comment (trunci) [reply lib/components/ReliefActivitiesMenu/ActivityButton.dart:148]: Maybe "Enter what mood you need to be in to complete this activity"?

File: lib/components/ReliefActivitiesMenu/ActivityButton.dart (hunk 2 above)
Comment (trunci) [reply lib/components/ReliefActivitiesMenu/ActivityButton.dart:203]: WDYM? The idea is that the URL is to an mp4/any video file.

**Comment by rithraj** [comment]
Fixed the conflict that was happening, I'm going to go ahead and merge it in now!

**Pull request merged**